### PR TITLE
Search for the events that start from...to

### PR DESCRIPTION
**Pull request by vubo**
Issue https://github.com/systers/vms/issues/81

Add a calender / date range opportunity for the event search.

Volunteer can find all events for specific dates.

image

### CODE REVIEW

File: vms/event/forms.py [hunk 1]
```diff
@@ -29,3 +30,8 @@ def clean(self):
                 self._errors['start_date'] = self.error_class([msg])
 
         return self.cleaned_data
+
+
+class EventDateForm(forms.Form):
```
Comment (vubo) [inline vms/event/forms.py]: Create a new form

File: vms/event/services.py [hunk 2]
```diff
@@ -50,6 +50,22 @@ def get_event_by_id(event_id):
     return result
 
 
+def get_events_by_date(start_date, end_date):
```
Comment (vubo) [inline vms/event/services.py]: Create a new function. Resut is an event list ordered by date.

File: vms/event/tests.py [hunk 3]
```diff
@@ -113,6 +119,55 @@ def test_get_event_by_id(self):
         self.assertNotEqual(get_event_by_id(200), e3)
         self.assertNotEqual(get_event_by_id(300), e3)
 
+    def test_get_events_by_date(self):
```
Comment (vubo) [inline vms/event/tests.py]: Create a new unit test for the function get_events_by_date

File: vms/event/views.py [hunk 4]
```diff
@@ -84,9 +86,30 @@ def list(request):
 
 @login_required
 def list_sign_up(request, volunteer_id):
```
Comment (vubo) [inline vms/event/views.py]: Extend the view

Comment (vubo) [inline vms/event/views.py]: Extend the view

**Comment by tapasweni-pathak** [comment]
@vubo What if user enters start date to be more then the end date. Can we give an error *Invalid Date Range* in such case?

**Comment by vubo** [comment]
@tapasweni-pathak user can not do it. Datepicker will not allow this.

image

**Comment by tapasweni-pathak** [comment]
Cool. :+1:

**Comment by willingc** [comment]
@vubo and @tapasweni-pathak You are both doing great work! :cake: :ocean: :bouquet:

**Comment by tapasweni-pathak** [comment]
@vubo is indeed doing great work. :+1: 

@willingc We all are awesome. :smile: :dancer:

**Comment by vubo** [comment]
@tapasweni-pathak @willingc THANK YOU :sunny:

**Pull request merged**